src/store/index.ts

```diff
@@ -1,8 +1,117 @@
 import { createStore } from "vuex";
+import Book from "@/model/book";
+
+import { format, parseISO } from "date-fns";
 
 export default createStore({
-  state: {},
-  mutations: {},
-  actions: {},
+  state: {
+    books: [] as Book[],
+    booksPublishedByMonth: {
+      january: {
+        amount: 0 as number,
+      },
+      february: {
+        amount: 0 as number,
+      },
+      march: {
+        amount: 0 as number,
+      },
+      april: {
+        amount: 0 as number,
+      },
+      may: {
+        amount: 0 as number,
+      },
+      june: {
+        amount: 0 as number,
+      },
+      july: {
+        amount: 0 as number,
+      },
+      august: {
+        amount: 0 as number,
+      },
+      september: {
+        amount: 0 as number,
+      },
+      october: {
+        amount: 0 as number,
+      },
+      november: {
+        amount: 0 as number,
+      },
+      december: {
+        amount: 0 as number,
+      },
+    },
+  },
+  mutations: {
+    addBook: (state, book) => {
+      state.books.push(book);
+    },
+    addBookAmountPublishedByMonth: (state, book) => {
+      switch (format(parseISO(book.publicationDate), "MM")) {
+        case "01":
+          state.booksPublishedByMonth.january.amount++;
+          break;
+        case "02":
+          state.booksPublishedByMonth.february.amount++;
+          break;
+        case "03":
+          state.booksPublishedByMonth.march.amount++;
+          break;
+        case "04":
+          state.booksPublishedByMonth.april.amount++;
+          break;
+        case "05":
+          state.booksPublishedByMonth.may.amount++;
+          break;
+        case "06":
+          state.booksPublishedByMonth.june.amount++;
+          break;
+        case "07":
+          state.booksPublishedByMonth.july.amount++;
+          break;
+        case "08":
+          state.booksPublishedByMonth.august.amount++;
+          break;
+        case "09":
+          state.booksPublishedByMonth.september.amount++;
+          break;
+        case "10":
+          state.booksPublishedByMonth.october.amount++;
+          break;
+        case "11":
+          state.booksPublishedByMonth.november.amount++;
+          break;
+        case "12":
+          state.booksPublishedByMonth.december.amount++;
+          break;
+        default:
+          return;
+      }
```

Comment:
```suggestion
    addBookCount: (state, book) => {
    	const month = format(parseISO(book.publicationDate), "M");
    	state.bookCountByMonth[month] = (state.bookCountByMonth[month] || 0) + 1;
    }
```

I think it's redundant to write out month names and have such a big switch case with only month indexes. We can make this much more generic and cleaner this way